### PR TITLE
[WIP] Technology Refactor

### PR DESCRIPTION
**Pull request by thomasdorch**
The aim of this PR is to combine layer display properties into a single module so that layouts have a more similar appearance between different visualization programs. Eventually, the `LayerView` class will replace `LayerColor` class, since it's basically the same thing, but with more functionality. 

- [x] Generate layer display properties (LayerViews) from a .yaml file, .lyp file, or python class and export to the same
- [x] `LayerColor`/`LayerBrightness`: Combine fill/frame values in yaml output if they are the same (see `layers.yaml`)
- [x] Move generic PDK into its own submodule
- [x] Create new technology module to contain all technology-related stuff in one place
- [x] Define LayerStack, LayerViews, and LayerMap in a more similar way
- [x] Move code-based utilities to `utils` submodule (`xml_utils`,` yaml_utils`, `color_utils`, `file_utils`, `function_utils`, and `async_utils`) 
- [x] Use `fill`/`frame` color in matplotlib/holoviews
- [ ] Use `line_style`, `dither_pattern`  in matplotlib/holoviews

@joamatab
@tvt173

### CODE REVIEW

**Comment by joamatab** [comment]
Looks great 😊 thank you Thomas

**Comment by joamatab** [comment]
see https://github.com/gdsfactory/gdsfactory/pull/1030

**Comment by thomasdorch** [comment]
Hi @joamatab 

After starting down the path to replace `LayerColors` with `LayerViews`, I figured it would make sense to just do a larger PR including a good deal of refactoring the way we handle technologies. 
One of the main issues I find is that the `LayerStack`, `LayerColors`, and `LayerMap` are all very closely related, but end up being defined very differently and were located in different places. To make it simpler, I created a new module (`technology`) for us to put things related to technologies (physical layer properties, layer display properties, simulation settings, etc.) and moved everything associated with the _generic_ technology into `generic_tech`.
Check out the `04_pdk.ipynb` notebook to see the changes in action!

-Thomas

File: gdsfactory/simulation/devsim/get_simulation.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def create_2Duz_simulation(
 if __name__ == "__main__":
 
     import gdsfactory as gf
-    from gdsfactory.tech import get_layer_stack_generic
+    from gdsfactory.generic_tech import get_layer_stack_generic
```
Comment (joamatab) [inline gdsfactory/simulation/devsim/get_simulation.py:240]: great!

**Comment by joamatab** [comment]
Hi Thomas, this is a big PR and is starting to have merge conflicts,

I think we should try to merge it as soon as possible

we can have following PRs later on

**Comment by thomasdorch** [comment]
Agreed, I'll have time tonight to work on this and get those conflicts resolved.

**Comment by joamatab** [comment]
I fixed the conflicts in this PR, thank you Thomas!
it looks great

https://github.com/gdsfactory/gdsfactory/pull/1102

**Pull request closed without merging**